remix-solidity/src/compiler/compiler-worker.ts

```diff
@@ -1,26 +1,23 @@
 'use strict'
 
-const solc = require('solc/wrapper')
-
-let compileJSON = function () { return '' }
-const missingInputs = []
-
-module.exports = function (self) {
-  self.addEventListener('message', (e) => {
-    const data = e.data
+import solc from 'solc/wrapper'
+import { CompilerInput, MessageToWorker } from './types'
+var compileJSON: ((input: CompilerInput) => string) | null = (input) => { return '' }
+var missingInputs: string[] = []
+
+export default (self) => {
```

Comment:
What the type of `self` ?
Also you can do that if you want : 
```typescript
function myMethod(this: MyClass, name: string) {
  this.name = name;
}
class MyClass {
  method = myMethod.bind(this);
  name: string
}
const myClass = new MyClass();
myClass.method('test');
console.log(myClass.name); // 'test'
```
I think it's nice way of dealing with composition.

Reply:
self is of type 'DedicatedWorkerGlobalScope' which is under 'webworker' library file. But 'dom' & 'webworker' library can't coexist. See: https://github.com/microsoft/TypeScript/issues/20595